### ISSUE TITLE
Remove `block_content_permissions` module

### ISSUE THREAD

**Issue by pyrello** [uiowa/uiowa#7129]
# Motivation

The `block_content_permissions` module is no longer needed after upgrading to 10.1 because the functionality exists in core.

# Proposed solution

```
ddev composer remove drupal/block_content_permissions
```

**Comment by pyrello** [comment]
This is blocked until after #7098 is merged and deployed.

**Comment by joewhitsitt** [comment]
no longer blocked
